foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/internal/oxm/XMLCompositeCollectionMappingNodeValue.java

```diff
@@ -93,18 +93,15 @@ public boolean marshal(XPathFragment xPathFragment, MarshalRecord marshalRecord,
         if (null != iterator && cp.hasNext(iterator)) {
             XPathFragment groupingFragment = marshalRecord.openStartGroupingElements(namespaceResolver);
             marshalRecord.closeStartGroupingElements(groupingFragment);
-        } else {
-            return marshalRecord.emptyCollection(xPathFragment, namespaceResolver, xmlCompositeCollectionMapping.getWrapperNullPolicy() != null);
-        }
-
-        marshalRecord.startCollection();
-        iterator = cp.iteratorFor(collection);
-        while (cp.hasNext(iterator)) {
-            Object objectValue = cp.next(iterator, session);
-            marshalSingleValue(xPathFragment, marshalRecord, object, objectValue, session, namespaceResolver, ObjectMarshalContext.getInstance());
+            marshalRecord.startCollection();
+            while (cp.hasNext(iterator)) {
```

Comment:
`cp.hasNext()` is being called twice in a row but it is clear that there will be at least one element already. Looks like good place to use `do { ... } while (cp.hasNext());` instead of `while (...) { ...}`